File: app/src/main/java/fr/frogdevelopment/pregnancycalendar/data/Contraction.java
package fr.frogdevelopment.pregnancycalendar.data;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.time.ZonedDateTime;

@Entity
public class Contraction {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    public Integer id;

    @ColumnInfo(name = "datetime")
    public ZonedDateTime dateTime;

    @ColumnInfo
    public Long duration;

}

File: app/src/main/java/fr/frogdevelopment/pregnancycalendar/data/Contraction.kt
package fr.frogdevelopment.pregnancycalendar.data

import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import org.threeten.bp.ZonedDateTime

@Entity(tableName = "CONTRACTION")
class Contraction {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    var id: Int? = null

    @NonNull
    @ColumnInfo(name = "DATETIME")
    var dateTime: ZonedDateTime? = null

    @NonNull
    @ColumnInfo(name = "DURATION")
    var duration: Long? = null
}
